docs: add filesystems model to reference (#191)
canonical/imagecraft: docs/reference/imagecraft-yaml-file.rst

.. kitbash-field:: StructureItem filesystem_label
    :prepend-name: volumes.<volume-name>.structure.<partition>

Filesystem keys
---------------

The following keys can be declared for each filesystem mount listed.

.. py:currentmodule:: craft_parts.filesystem_mounts

.. kitbash-field:: FilesystemMountItem mount

>     :prepend-name: filesystems.<filesystem-name>.<filesystem>

Reviewer: `<filesystem-mount>` feels redundant given that its parent is a named filesystem. I lean towards `<mount>` because the hierarchy already asserts that it belongs to a filesystem. It's also consistent with how the `volumes` child keys are titled.

.. kitbash-field:: StructureItem filesystem_label
    :prepend-name: volumes.<volume-name>.structure.<partition>

Filesystem keys
---------------

The following keys can be declared for each filesystem mount listed.

.. py:currentmodule:: craft_parts.filesystem_mounts

.. kitbash-field:: FilesystemMountItem mount
    :prepend-name: filesystems.<filesystem-name>.<filesystem>

>     Mountpoint to mount the device at.

Reviewer: ```suggestion
    The device's mount point.
```

**Examples**

.. code-block:: yaml

    mount: "/"

.. code-block:: yaml

    mount: "/boot/efi"

.. kitbash-field:: FilesystemMountItem device
    :prepend-name: filesystems.<filesystem-name>.<filesystem>

>     Device to be mounted. Must reference a partition.

Reviewer: Sorry about this. I had only planned on this being used to supplement the description, so I didn't account for the case where there is no field description. I created an [issue](https://github.com/canonical/pydantic-kitbash/issues/56) to track this.

In the meantime, we can just add these in with rST.

```suggestion
**Description**

Device to be mounted. Must reference a partition.
```

We do the same thing in Snapcraft for application-specific behavior.

Reviewer: Suggestion for the description itself:

```suggestion
The device to be mounted. This must reference one of the partitions defined in `volumes.<volume-name>.structure`.
```
